Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -463,6 +463,15 @@ public class PBXProjGenerator {
             }
             return !linkingAttributes.isEmpty ? ["ATTRIBUTES": linkingAttributes] : nil
         }
+        
+        func isStaticDependency(for carthageDependency: Dependency) -> Bool {
```

Comment:
I agree about moving this to an extension. If we move static/linkType to Dependency itself like in the other comment we don't even need this function. 

Otherwise if this crashes when used with other types it should be called `carthageDependencyIsStatic`, otherwise it should probably just return false for other cases